resources/resources.go

```diff
@@ -68,8 +68,12 @@ func (res *EC2NomadResource) Scale(desiredNomadCount int, vc *VaultClient) error
 	if count, err := res.NomadClient.GetTaskGroupCount(); err == nil {
 		if count < desiredNomadCount { // scale out
 			return res.scaleOut(desiredNomadCount, vc)
+		} else if count < res.NomadClient.MinCount {
+			return res.scaleOut(res.NomadClient.MinCount, vc)
 		} else if count > desiredNomadCount {
 			return res.scaleIn(desiredNomadCount, vc)
+		} else if count > res.NomadClient.MaxCount {
+			return res.scaleIn(res.NomadClient.MaxCount, vc)
```

Comment:
Two points:

1. There's no guarantee that `res.NomadClient.MinCount <= desiredNomadCount <= res.NomadClient.MaxCount`. Either we check for this condition first, return error if not fulfilled (preferred), or we just skip this step and go to point 2.
2. The logic is easier to follow if we bunch together the if-else `< desiredNomadCount` and `> desiredNomadCount` first, then followed by the `< res.NomadClient.MinCount` and `> res.NomadClient.MaxCount`. In fact when paired with point 1, it is possible to just scale out to max(`desiredNomadCount`, `res.NomadClient.MinCount`), and scale in to min(`desiredNomadCount`, `res.NomadClient.MaxCount`), though Golang standard lib doesn't have max and min for integers zzz.

Reply:
for point 1, if `res.NomadClient.MinCount > desiredNomadCount` or `desiredNomadCount > res.NomadClient.MaxCount`, there shouldn't be an error, instead it should scale accordingly to meet the new boundary. 

e.g. if the count currently is at 30, `min=10 max=40` and the user decided that the max should be 20 instead, the scaler should not throw error when `current > max`, instead it should scale `current` to `max`. Likewise for the case where `current < min`.

Reply:
for point 2, i think there isn't a point in a ` max(desiredNomadCount, res.NomadClient.MinCount)`

if `count < desiredNomadCount` is false and `count < res.NomadClient.MinCount` is true, it means `res.NomadClient.MinCount > desiredNomadCount`. Unless, we do 

```
if count < desiredNomadCount || count < res.NomadClient.minCount {
    scale to max(desiredNomadCount, res.NomadClient.MinCount)
}

```